### PR TITLE
add status to Task

### PR DESCRIPTION
**Pull request by jakobroehrl**
Signed-off-by: Jakob <jakob.roehrl@web.de>
I'm a little bit proud, this is my first Task Issue without any help from others :)
It solves #8 

@jancborchardt Do you have a nice icon for me? Currently I copied the status icon, thanks.
grafik

### CODE REVIEW

File: src/store/tasks.js [hunk 1]
```diff
@@ -419,6 +419,17 @@ const mutations = {
 		Vue.set(task, 'class', classification)
 	},
 
+	/**
+	 * Sets the classification of a task
```
Comment (raimund-schluessler) [inline src/store/tasks.js]: Should be 'status' instead of 'classification'.

File: src/store/tasks.js [hunk 2]
```diff
@@ -419,6 +419,17 @@ const mutations = {
...
+	 *
+	 * @param {Object} state The store data
+	 * @param {Task} task The task
+	 * @param {String} classification The classification
```
Comment (raimund-schluessler) [inline src/store/tasks.js]: Same here.

File: src/store/tasks.js [hunk 3]
```diff
@@ -953,6 +964,19 @@ const actions = {
...
 	},
 
+	/**
+	 * Sets the status of a task
+	 *
+	 * @param {Object} context The store context
+	 * @param {Task} task The task to update
+	 */
+	async setStatus(context, { task, status }) {
+		// check classification to comply with RFC5545 values
```
Comment (raimund-schluessler) [inline src/store/tasks.js]: status?

File: src/components/TheDetails.vue [hunk 4]
```diff
@@ -529,6 +552,12 @@ export default {
 				{ displayName: this.$t('tasks', 'When shared show only busy'), type: 'CONFIDENTIAL' },
 				{ displayName: this.$t('tasks', 'When shared hide this event'), type: 'PRIVATE' },
 			],
+			statusSelect: [
+				{ displayName: this.$t('tasks', 'needs action'), type: 'NEEDS-ACTION' },
```
Comment (raimund-schluessler) [inline src/components/TheDetails.vue]: Better capitalize the beginning to match with the others, e.g. class.

**Comment by raimund-schluessler** [comment]
Looks good now, I will test asap. Still needs a better icon though.

**Comment by jancborchardt** [comment]
> I'm a little bit proud, this is my first Task Issue without any help from others :)

Also congrats @jakobroehrl! :tada:  :)

**Comment by jakobroehrl** [comment]
@raimund-schluessler 
How can I get the svg from the icon? On the manuel is only the png.

**Comment by raimund-schluessler** [comment]
> Also sorting-wise, I feel it belongs to "Priority" or completion %, what do you think?

I would put it above the piority, below the class selection (one row down).

> Icon-wise, I would propose icon-category-monitoring:

We have this icon in the repo already: https://github.com/nextcloud/tasks/blob/master/img/src/bw/current.svg You can simply apply the classes `icon icon-bw icon-current`. But we might want to have it in blue (just copy it to https://github.com/nextcloud/tasks/tree/master/img/src/color, adjust the color, run the `npm run svg_sprites` and apply the classes `icon icon-color icon-current`).

**Comment by jakobroehrl** [comment]
changed the ordering.
The black icon is working, but with color I get an odd one:
grafik

**Comment by raimund-schluessler** [comment]
> The black icon is working, but with color I get an odd one:

Did you run `npm run svg_sprite`? You might also try to clear the browser cache. And check whether you see the icon in `nextcloud/apps/tasks/img/color.svg`

**Comment by jakobroehrl** [comment]
> 
> 
> > The black icon is working, but with color I get an odd one:
> 
> Did you run `npm run svg_sprite`? You might also try to clear the browser cache. And check whether you see the icon in `nextcloud/apps/tasks/img/color.svg`

Yes, I run npm run svg_sprite and cleared the cache. But I only see one current image in nextcloud/apps/tasks/img/color.svg. It's not my blue one. Should we simply use it in black?

**Comment by raimund-schluessler** [comment]
Then there is something wrong. I will have a look in the evening.

**Comment by raimund-schluessler** [comment]
I fixed the issue with the icon in ad112c5. Now looks like this if no status is set:
Screenshot_2019-10-09 Aufgaben - Nextcloud

and like this when a status is there:
Screenshot_2019-10-09 Aufgaben - Nextcloud(1)

Feel free to merge.

**Comment by jakobroehrl** [comment]
> 
> 
> One thing that just came to my mind: When we change the complete value of a task, whe change the status accordingly: https://github.com/nextcloud/tasks/blob/master/src/models/task.js#L242-L257
> complete 0 % => NEEDS-ACTION,
> complete > 0 % => IN-PROCESS,
> complete 100 % => COMPLETED
> 
> I think we should we vice-versa also set the complete value when we adjust the status manually. At the moment it is a bit weird e.g. if you have a completed task and manually set the status to IN-PROCESS. We might want to set the complete value to e.g. 99 % then.
> 
> So:
> NEEDS-ACTION: complete = 0
> IN-PROCESS: complete > 0, < 100,
> COMPLETED: complete 100,
> CANCELED: leave the complete value as is.
> 
> What do you think?

Let's merge it and do this maybe later, OK?

How can I merge? Haven't done this yet.

**Comment by raimund-schluessler** [comment]
> Let's merge it and do this maybe later, OK?

Fine with. Could you just create an issue, so we don't forget?

The "Merge pull request" button is down below on the left 😉

**Pull request merged**